extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/HttpUpgradeContextImpl.java

```diff
@@ -1,14 +1,22 @@
 package io.quarkus.websockets.next.runtime;
 
+import java.util.Map;
+
 import io.quarkus.security.identity.SecurityIdentity;
 import io.quarkus.websockets.next.HttpUpgradeCheck;
+import io.quarkus.websockets.next.UserData;
 import io.smallrye.mutiny.Uni;
 import io.vertx.core.http.HttpServerRequest;
 import io.vertx.ext.web.RoutingContext;
 
-record HttpUpgradeContextImpl(RoutingContext routingContext,
+record HttpUpgradeContextImpl(RoutingContext routingContext, UserData userData,
         Uni<SecurityIdentity> securityIdentity, String endpointId) implements HttpUpgradeCheck.HttpUpgradeContext {
 
+    @Override
+    public Map<String, String> pathParams() {
+        return routingContext.pathParams();
```

Comment:
I think that the map returned from `routingContext.pathParams()` is mutable so we should either use `String pathParam(String name)` (see my comment in the `HttpUpgradeCheck`) or use an immutable copy of this map (i.e. something like `Map.copyOf(routingContext.pathParams())`).